langchain/document_loaders/srt.ts

```diff
@@ -0,0 +1,59 @@
+import type { readFile as ReadFileT } from "fs/promises";
+import type SRTParserT from "srt-parser-2";
+import { Document } from "../document";
+import { BaseDocumentLoader } from "./base";
+
+let readFile: typeof ReadFileT | null = null;
+let SRTParser: typeof SRTParserT | null = null;
+
+try {
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  ({ readFile } = require("fs/promises"));
+  // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+  SRTParser = require("srt-parser-2");
+} catch {
+  // ignore error, will be throw in constructor
+}
+
+export class SRTLoader extends BaseDocumentLoader {
+  constructor(public filePath: string) {
+    super();
+
+    /**
+     * Throw error at construction time
+     * if fs/promises is not installed.
+     */
+    if (readFile === null) {
+      throw new Error("Failed to load fs/promises.`");
```

Comment:
Would probably be good to throw a more informative error, like

```js
if (readFile === null) {
  const { isBrowser, isNode, isWebWorker, isJsDom, isDeno } = require('browser-or-node');
  let env: string;
  if (isBrowser) {
    env = 'browser';
  } else if (isNode) {
    env = 'node';
  } else if (isWebWorker) {
    env = 'webworker';
  } else if (isJsDom) {
    env = 'jsdom';
  } else if (isDeno) {
    env = 'deno';
  }
  throw new Error(`Failed to load fs/promises. SRTLoader available only on environment 'node'. It appears you are running environment '${env}'. See https://<link to docs> for alternatives.`);
}
```

(obviously we'd want to encapsulate this in a util function we can use in multiple places